discussions/tasks.py

```diff
@@ -106,25 +106,49 @@ def sync_channel_memberships():
 def add_user_as_moderator_to_channel(user_id, program_id):
     """
     Add moderator to a open-discussions channels of given program
+
+    Args:
+        user_id (int): user id
+        program_id(int): program id
     """
     if not settings.FEATURES.get('OPEN_DISCUSSIONS_USER_SYNC', False):
         log.debug('OPEN_DISCUSSIONS_USER_SYNC is set to False (so disabled) in the settings')
         return
 
-    channel_program_filter = ChannelProgram.objects.filter(program_id=program_id)
-    for channel_program in channel_program_filter:
-        api.add_and_sub_moderator_to_channel(user_id, channel_program.channel.name)
+    try:
+        discussion_user = DiscussionUser.objects.get(user_id=user_id)
+        channel_programs = ChannelProgram.objects.filter(program_id=program_id)
+
+        for channel_program in channel_programs:
+            api.add_and_subscribe_moderator(
+                discussion_user.username,
+                channel_program.channel.name
+            )
+    except DiscussionUser.DoesNotExist:
```

Comment:
Yes, but we typically follow a best practice of scoping the exception handling only to the line that will raise it. The reason being is that you can't assume something like `remove_moderator_from_channel` will never raise that error in the future if someone separately changes that implementation. Another way to do this would be to `filter().first()` and check for `is not None` to execute this code.

Reply:
ok